src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -658,6 +658,10 @@ public virtual void Generate(NpgsqlDropDatabaseOperation operation, [CanBeNull]
             Check.NotNull(operation, nameof(operation));
             Check.NotNull(builder, nameof(builder));
 
+            // Alter operations may not have a default schema attached. If not, try to attach one for schema-qualified types.
+            if (operation[RelationalAnnotationNames.DefaultSchema] == null)
```

Comment:
I'm not too familiar with EF Core internals, so I wonder why in some places we check for an empty string and not in other.

Reply:
Good question... I've seen the null checks in practice, but I've been adding the empty checks where it's obvious that an empty string would lead to malformed SQL.

@roji Any thoughts on this? Is there a precondition that obviates the need for later empty checks, or should these empty checks be the norm?

Reply:
Nothing in particular to say, I guess it's on a case-by-case basis? Default schema seems like a clear case where an empty string (as opposed to null) should not be legal, maybe in certain other cases it does make sense?

(will do my best to give this PR a general review today/tomorrow)